prov/gni/src/gnix_msg.c

```diff
@@ -923,13 +1460,37 @@ static int __smsg_rndzv_start(void *data, void *msg)
 		req->msg.rndzv_head = hdr->head;
 		req->msg.rndzv_tail = hdr->tail;
 
+		if (req->type == GNIX_FAB_RQ_RECV) {
+			/* fi_send is rndzv with recv */
+
+			GNIX_DEBUG(FI_LOG_EP_DATA, "recv rndzv start - POSTED,"
+				   " req = %p\n", req);
+			req->work_fn = __gnix_rndzv_req;
+		} else if (req->type == GNIX_FAB_RQ_RECVV ||
+			   req->type == GNIX_FAB_RQ_TRECVV) {
+			/* fi_send is rndzv with recvv */
+
+			GNIX_DEBUG(FI_LOG_EP_DATA, "recvv rndzv start - POSTED,"
+				   " req = %p\n", req);
+
+			req->work_fn = __gnix_rndzv_iov_req_build;
+			_gnix_remove_tag(posted_queue, req);
+			ret = _gnix_vc_queue_work_req(req);
+
+			GNIX_DEBUG(FI_LOG_EP_DATA,
+				   "_gnix_vc_queue_work_req returned %s\n",
+				   fi_strerror(-ret));
+			goto fini;
```

Comment:
I'd really prefer that you avoid using goto.  It looks like it's only preventing RECVV from supporting MULTI_RECV.  You could achieve the same by testing for RECV in the MULTI_RECV test just below, like:

```
 		if (req->type == GNIX_FAB_RQ_RECV && (req->msg.recv_flags & FI_MULTI_RECV) &&
		    ((req->msg.cum_recv_len - req->msg.cum_send_len) >= ep->min_multi_recv)) {
```